tests/ttnn/unit_tests/operations/test_distributed_layernorm_sharded.py

```diff
@@ -0,0 +1,472 @@
+
+    # shard to 1 core
+    tt_stats_sharded_config = ttnn.create_sharded_memory_config(
+        shape=(1, 1, 32, tt_stats_tensor.get_legacy_shape()[-1]),
+        core_grid=ttnn.CoreGrid(y=1, x=1),
+        strategy=ttnn.ShardStrategy.WIDTH,
+    )
+    tt_stats_tensor = ttnn.to_memory_config(tt_stats_tensor, memory_config=tt_stats_sharded_config)
+
+    for iter in range(iterations):
+        if is_rmsnorm:
+            tt_output_tensor = ttnn.rms_norm_post_all_gather(
+                tt_input_tensor,
+                epsilon=eps,
+                weight=tt_weights,
+                program_config=SHARDED_NORM_PRGM_CFG,
+                memory_config=tt_sharded_config,
+                stats=tt_stats_tensor,
+            )
+        else:
+            tt_output_tensor = ttnn.layer_norm_post_all_gather(
+                tt_input_tensor,
+                epsilon=eps,
+                weight=tt_weights,
+                program_config=SHARDED_NORM_PRGM_CFG,
+                memory_config=tt_sharded_config,
+                stats=tt_stats_tensor,
+            )
+        tt_output_torch = ttnn.to_torch(tt_output_tensor).to(torch.bfloat16)
+        _, pcc_out = comp_pcc(torch_output_tensor, tt_output_torch, pcc=min_pcc)
+        all_close_passing = torch.allclose(torch_output_tensor, tt_output_torch, atol=max_atol, equal_nan=False)
+        atol_delta = torch.max(torch.abs(torch_output_tensor - tt_output_torch)).item()
+        assert pcc_out >= min_pcc, f"PCC test failed: {pcc_out} (threshold: {min_pcc})"
+        assert atol_delta <= max_atol, f"Max Atol exceeded: {atol_delta} (allowed: {max_atol})"
+
+
+@skip_for_grayskull()
+@pytest.mark.parametrize("is_rmsnorm", [True, False])  # Layernorm not supported for now
+@pytest.mark.parametrize("seed", [0, 1234])  # Test across 5 different seeds
+@pytest.mark.parametrize("eps", [1e-6])
+@pytest.mark.parametrize("min_pcc", [0.9997])
+@pytest.mark.parametrize("max_atol", [0.38])
+@pytest.mark.parametrize("input_width", [2048])
+@pytest.mark.parametrize("num_devices", [4, 8])
+@pytest.mark.parametrize(
+    "input_df",
+    [ttnn.bfloat8_b, ttnn.bfloat16],
+)
+@pytest.mark.parametrize(
+    "weights_df",
+    [ttnn.bfloat8_b, ttnn.bfloat16],
+)
+@pytest.mark.parametrize(("mean", "std"), ([0, 1],))
+@pytest.mark.parametrize(
+    "core_grid",
+    ((4, 8),),
+)
+def test_post_allgather_layernorm(
+    all_devices,
+    use_program_cache,
+    input_width,
+    num_devices,
+    is_rmsnorm,
+    input_df,
+    weights_df,
+    seed,
+    eps,
+    mean,
+    std,
+    min_pcc,
+    max_atol,
+    core_grid,
+):
+    device = all_devices[0]
+    run_post_allgather_layernorm(
+        device,
+        input_width,
+        num_devices,
+        is_rmsnorm,
+        input_df,
+        weights_df,
+        seed,
+        eps,
+        mean,
+        std,
+        min_pcc,
+        max_atol,
+        core_grid,
+        2,
```

Comment:
Would be nice to have iterations=2 as a pytest param for better visibility. Same comment applies for test_pre_allgather_layernorm